test/integration/examples/apiserver_test.go

```diff
@@ -334,9 +337,8 @@ func TestAggregatedAPIServer(t *testing.T) {
 	// this is ugly, but sleep just a little bit so that the watch is probably observed.  Since nothing will actually be added to discovery
 	// (the service is missing), we don't have an external signal.
 	time.Sleep(100 * time.Millisecond)
-	if _, err := aggregatorDiscoveryClient.Discovery().ServerResources(); err != nil {
-		t.Fatal(err)
-	}
+	_, err = aggregatorDiscoveryClient.Discovery().ServerResources()
+	assertWardleUnavailableDiscoveryError(t, err)
```

Comment:
The wardle `APIService` was never available, because `Service` object is missing, which is referenced in the `APIService` object above:
```go
 			Service: &apiregistrationv1beta1.ServiceReference{
 				Namespace: "kube-wardle",
 				Name:      "api",
},
```
Kube-aggregator previously silently skipped unavailable API services in the `/apis` endpoint. With the new change, there is an error with details that are asserted in the `assertWardleUnavailableDiscoveryError` function (since `.Discovery().ServerResources()` now invokes `/apis/wardle.k8s.io/v1alpha1` endpoint that returns `503` error).

Reply:
I think it's good to verify this error this way to make sure discovery reports it is incomplete in this case. once we verify this way, can we create the service pointing at `127.0.0.1:<wardlePort>` and make sure we get a successful discovery?

Reply:
actually, I'm not sure we have the network infrastructure here to use services

Reply:
heh, your initial suggestion made me wonder if there was some magic trick to make such Service